typing/typedecl.ml

```diff
@@ -544,7 +544,7 @@ let rec check_constraints_rec env loc visited ty =
            *already* violate the constraints -- we need to report a problem with
            the unexpanded types, or we get errors that talk about the same type
            twice.  This is generally true for constraint errors. *)
-        try Ctype.matches ~expand_error_trace:false env ty ty'
+        try Ctype.matches ~expand_error_trace:false env (Ctype.instance ty) ty'
```

Comment:
This feels like an impedance mismatch since `Ctype.matches` starts with taking a snapshot and reverts to this former state before exiting. Thus taking an instance sounds like unnecessary work in this context.

Reply:
Yes, you're right! I retract my PR.